source/core/serial_solid.cc

```diff
@@ -104,27 +98,58 @@ std::vector<
   std::pair<typename Triangulation<spacedim>::active_cell_iterator,
             typename Triangulation<dim, spacedim>::active_cell_iterator>>
 SerialSolid<dim, spacedim>::map_solid_in_background_triangulation(
-  const parallel::distributed::Triangulation<spacedim> &background_tr,
-  const std::shared_ptr<Triangulation<dim, spacedim>> & solid_tr)
+  const parallel::TriangulationBase<spacedim> &background_tr)
 {
   std::vector<
     std::pair<typename Triangulation<spacedim>::active_cell_iterator,
               typename Triangulation<dim, spacedim>::active_cell_iterator>>
     mapped_solid;
 
+
+  // Gather the amount of vertices per cell in the solid cells once to have
+  // static memory
+  auto                         temporary_solid_cell = solid_tria->begin();
+  std::vector<Point<spacedim>> triangle(temporary_solid_cell->n_vertices());
+
+
+  // Calculate distance from cell center to  solid_cell
```

Comment:
```suggestion
  // Calculate distance from cell center to solid_cell
```

Reply:
Cell center to solid cell face/surface ?

Reply:
Well should be triangle